ui/apps/everest/src/pages/db-cluster-details/cluster-overview/cards/resources-details.tsx

```diff
@@ -13,50 +13,161 @@
+    customNrOfProxies,
+  }) => {
+    updateDbClusterResources(
+      {
+        dbCluster,
+        newResources: {
+          cpu,
+          disk,
+          diskUnit,
+          memory,
+          proxyCpu,
+          proxyMemory,
+          numberOfProxies: parseInt(
+            numberOfProxies === CUSTOM_NR_UNITS_INPUT_VALUE
+              ? customNrOfProxies || '1'
+              : numberOfProxies,
+            10
+          ),
+          numberOfNodes: parseInt(
+            numberOfNodes === CUSTOM_NR_UNITS_INPUT_VALUE
+              ? customNrOfNodes || '1'
+              : numberOfNodes,
+            10
+          ),
+        },
+      },
+      {
+        onSuccess: () => {
+          setOpenEditModal(false);
+          queryClient.invalidateQueries({
+            queryKey: [DB_CLUSTER_QUERY, dbCluster.metadata.name],
+          });
+        },
+      }
+    );
+  };
+
   return (
-    <OverviewCard
-      dataTestId="resources"
-      cardHeaderProps={{
-        title: Messages.titles.resources,
-        avatar: <DatabaseIcon />,
-        // TODO implement with EVEREST-1211
-        // action: (
-        //     <Button size="small" startIcon={<EditOutlinedIcon />}>
-        //     Edit
-        //   </Button>
-        // ),
-      }}
-    >
-      <OverviewSection
-        title={`${numberOfNodes} node${+numberOfNodes > 1 ? 's' : ''}`}
-        loading={loading}
+    <>
+      <OverviewCard
+        dataTestId="resources"
+        cardHeaderProps={{
+          title: Messages.titles.resources,
+          avatar: <DatabaseIcon />,
+          action: (
+            <Button
+              size="small"
+              startIcon={<EditOutlinedIcon />}
+              onClick={() => setOpenEditModal(true)}
+            >
+              Edit
+            </Button>
+          ),
+        }}
       >
-        <OverviewSectionRow
-          label={Messages.fields.cpu}
-          contentString={`${cpu}`}
-        />
-        <OverviewSectionRow
-          label={Messages.fields.disk}
-          contentString={`${disk}`}
-        />
-        <OverviewSectionRow
-          label={Messages.fields.memory}
-          contentString={`${memory}`}
+        <OverviewSection
+          title={`${replicas} node${+replicas > 1 ? 's' : ''}`}
+          loading={loading}
+        >
+          <OverviewSectionRow
+            label={Messages.fields.cpu}
+            contentString={`${cpu}`}
+          />
+          <OverviewSectionRow
+            label={Messages.fields.disk}
+            contentString={`${disk}`}
+          />
+          <OverviewSectionRow
+            label={Messages.fields.memory}
+            contentString={`${memory}`}
+          />
+        </OverviewSection>
+      </OverviewCard>
+      {openEditModal && (
+        <ResourcesEditModal
+          dbType={dbType}
+          handleCloseModal={() => setOpenEditModal(false)}
+          onSubmit={onSubmit}
+          defaultValues={{
+            cpu: cpuParser(cpu.toString() || '0'),
```

Comment:
What do you think about throwing dbCluster inside and decomposing the values directly related to the shape inside the component, so as not to expand the one component above? This is more a suggestion and we can think about it later